chatgpt-shell.el

```diff
@@ -1418,15 +1405,22 @@ enables additional key bindings.
           (chatgpt-shell--primary-buffer)
         (chatgpt-shell--put-source-block-overlays)))))
 
+(defun chatgpt-shell-ask-chatgpt-region ()
+  "Send text from region using ChatGPT"
+  (interactive)
+  (unless (region-active-p)
+    (user-error "No region active"))
+  (let* ((region-text (buffer-substring (region-beginning) (region-end)))
+         )
+    (chatgpt-shell-send-to-buffer trim-leading nil)))
+
 (defun chatgpt-shell-send-region-with-header (header)
   "Send text with HEADER from region using ChatGPT."
   (unless (region-active-p)
     (user-error "No region active"))
-  (chatgpt-shell-send-to-buffer
-   (concat header
-           "\n\n"
-           (buffer-substring (region-beginning) (region-end)))
-   nil))
+  (let* ((region-text (buffer-substring (region-beginning) (region-end)))
```

Comment:
Let's keep the previous version. We aren't getting additional benefits from a local var, but also lose the explicitness of what previous code did.

Reply:
hmm i am not quite agree with this one, cause nested expression might make debug / understanding difficult. breaking them into small piece might help understanding? my experience in other programming language let me made that change

Reply:
also, i am still on the road learning emacs lisp. there is a built in (interactive "p") for region actions. wonder if it is good to move to use that way for existing region related functions in future, for simplicity? i could be wrong.

Reply:
> nested expression might make debug / understanding / maintainance difficult.

The code we're considering here is trivial. Let's keep the previous please.

Reply:
OK, i can make it back